src/Nethermind/Nethermind.Serialization.Rlp/KeyValueStoreRlpExtensions.cs

```diff
@@ -49,42 +49,13 @@ public static class KeyValueStoreRlpExtensions
         {
             if (decoder is IRlpValueDecoder<TItem> valueDecoder)
             {
-                Span<byte> data = db.GetSpan(key);
-                if (data.IsNull())
-                {
-                    return null;
-                }
-
-                try
-                {
-                    if (data.Length == 0)
-                    {
-                        return null;
-                    }
-
-                    var rlpValueContext = data.AsRlpValueContext();
-                    item = valueDecoder.Decode(ref rlpValueContext, rlpBehaviors | RlpBehaviors.AllowExtraBytes);
-                }
-                finally
-                {
-                    db.DangerousReleaseMemory(data);
-                }
+                item = db is IReadOnlyNativeKeyValueStore native
+                    ? GetViaNative(native, key, valueDecoder, rlpBehaviors)
+                    : GetViaSpan(db, key, valueDecoder, rlpBehaviors);
```

Comment:
Could we utilize generics here?

Reply:
Don't think so as is pattern matching to get type.

Have changed to have same names though and just use overloading

Reply:
But you should have made those overloads public, they are still private

Reply:
Could also make the method calling them private as not called outside the extension